### PR TITLE
Fix can't "Select All" collection items if all items are pinned

### LINKED ISSUE #16497: Can't "Select All" collection items if all items are pinned
**Describe the bug**
On the collection page, if a user tries to select all items using the checkbox in the bulk actions bar, the selection doesn't work if all of the collection items are pinned.

**Logs**

Nothing specific to the problem

**To Reproduce**
1. Open a collection where every item is pinned
2. Hover the item's icon and click the checkbox that appeared
3. Click the checkbox in the bottom bulk actions bar
4. After clicking a checkbox, all items get unselected

**Expected behavior**
All the collection items have to be selected for a bulk operation

**Demo**

https://user-images.githubusercontent.com/17258145/121223188-f1f25080-c88f-11eb-9042-a51a7a56c209.mov

**Information about your Metabase Installation:**

Metabase version: latest master ([commit](https://github.com/metabase/metabase/commit/bbd4f635220b05487de82a18310d4d18fe04174d))
Metabase hosting environment: local, master branch
Metabase internal database: H2 (default), sample dataset

### PR DESCRIPTION
**Pull request by kulyk**
On the collection page, if a user tries to select all items using the checkbox in the bulk actions bar, the selection doesn't work if _all of the collection items are pinned_.

The issue is the `unchecked` list that influences the bulk operations behavior was created only out of non-pinned items. So if the collection only contains pinned items, you selected one of the items, the `unchecked` appeared to be empty, and the `BulkActions` component was sure there is nothing more to select.

- Closes #16497
- Adds repro for #16497 together with the fix

### To Verify

⚠️ You might try to archive selected items during testing. This behavior is currently broken, the fix is at #16500

1. Open a collection that contains ONLY pinned items
2. Hover the item's icon and click the checkbox that appeared
3. Click the checkbox in the bottom bulk actions bar
4. After clicking a checkbox, all items have to be selected

### Demo

**Before**

https://user-images.githubusercontent.com/17258145/121223188-f1f25080-c88f-11eb-9042-a51a7a56c209.mov

**After**

https://user-images.githubusercontent.com/17258145/121237590-e27b0380-c89f-11eb-90c7-8c302468ca04.mov

### CODE REVIEW

**Review by alxnddr — APPROVED** [review]
I missed that case, thanks for fixing 🥇

File: frontend/test/metabase/scenarios/collections/collections.cy.spec.js [hunk 1]
```diff
@@ -593,8 +593,6 @@ describe("scenarios > collection_defaults", () => {
       it("should be possible to apply bulk selection to items (metabase#14705)", () => {
         selectItemUsingCheckbox("Orders");
         cy.findByText("1 item selected").should("be.visible");
-        selectItemUsingCheckbox("Orders in a dashboard", "dashboard");
-        cy.findByText("2 items selected").should("be.visible");
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/collections/collections.cy.spec.js:597]: Explanation:
We were actually making this exact same check in the next repro (making sure pinned item can be checked).

File: frontend/test/metabase/scenarios/collections/collections.cy.spec.js [hunk 2]
```diff
@@ -689,3 +689,17 @@ function getSidebarCollectionChildrenFor(item) {
...
 }
+
+function pinAllRootItems() {
+  cy.request("GET", "/api/collection/root/items").then(resp => {
+    const ALL_ITEMS = resp.body.data;
+
+    ALL_ITEMS.forEach(({ model, id }, index) => {
+      if (model !== "collection") {
+        cy.request("PUT", `/api/${model}/${id}`, {
+          collection_position: index++,
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/collections/collections.cy.spec.js:688]: Order of items is not important so I simply used index.

File: frontend/test/metabase/scenarios/collections/collections.cy.spec.js [hunk 3]
```diff
@@ -594,45 +594,33 @@ describe("scenarios > collection_defaults", () => {
...
-
-        // Select all
-        cy.icon("dash").click();
-        cy.icon("dash").should("not.exist");
-        cy.findByText("4 items selected");
-
-        // Deselect all
-        cy.findByTestId("bulk-action-bar").within(() => {
-          cy.icon("check").click();
+      describe("selection", () => {
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/collections/collections.cy.spec.js]: This is already a good indicator that this should be a separate file, but we can follow up on that.
```
bulk actions
  selection
  move
  archive
```

**Review by nemanjaglumac — APPROVED** [review]
Tested locally and it works like a charm 🚀

Sorry for messing with the repros. 🤨

**Pull request merged**